monai/networks/blocks/upsample.py

```diff
@@ -83,13 +85,21 @@ def __init__(
         if up_mode == UpsampleMode.DECONV:
             if not in_channels:
                 raise ValueError(f"in_channels needs to be specified in the '{mode}' mode.")
+
+            if not kernel_size:
+                kernel_size = scale_factor
+            kernel_size_ = ensure_tuple_rep(kernel_size, spatial_dims)
+            padding = tuple((i-1)//2 for i in kernel_size_)
+
             self.add_module(
                 "deconv",
                 Conv[Conv.CONVTRANS, spatial_dims](
                     in_channels=in_channels,
                     out_channels=out_channels or in_channels,
-                    kernel_size=scale_factor_,
+                    kernel_size=kernel_size_,
                     stride=scale_factor_,
+                    padding=padding,
+                    output_padding=padding,
```

Comment:
these paddings couldn't pass the test case:
https://github.com/Project-MONAI/MONAI/blob/ba670c9ba78f9f4d76eb12b4d33f245b5774d462/tests/test_upsample_block.py#L45-L47 perhaps add a `'same'` padding mode?